### PR TITLE
github-actions: testing improvements to style/unit-tests

### PR DESCRIPTION
**Pull request by alecbcs**
Moving from full `depth: 0` clones to partial `tree:0` seems to save us about 30s on each clone in CI making prechecks faster to complete.

### CODE REVIEW

File: .github/workflows/valid-style.yml [hunk 1]
```diff
@@ -13,52 +13,63 @@ concurrency:
...
+    - name: Install Python Packages
+      run: |
+        pip install --upgrade pip setuptools
+        pip install -r .github/workflows/style/requirements.txt
+
+    - name: vermin (Spack's Core)
+      run: vermin --backport importlib --backport argparse --violations --backport typing -t=3.6- -vvv lib/spack/spack/ lib/spack/llnl/ bin/
+
+    - name: vermin (Repositories)
+      run: vermin --backport importlib --backport argparse --violations --backport typing -t=3.6- -vvv var/spack/repos
```
Comment (alalazo) [inline .github/workflows/valid-style.yml]: Why was this moved?

Comment (alecbcs) [reply .github/workflows/valid-style.yml]: I don't remember. I've reverted to its previous placement.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -12,6 +12,6 @@ updates:
       interval: "daily"
   # Requirements to run style checks
   - package-ecosystem: "pip"
-    directory: "/.github/workflows/style"
+    directory: "/.github/workflows/requirements"
```
Comment (alalazo) [inline .github/dependabot.yml:15]: Just a note that we might be relying on an undocumented feature: https://github.com/dependabot/dependabot-core/issues/3940#issuecomment-1227005267

Comment (alecbcs) [reply .github/dependabot.yml:15]: Hoping they add official documentation soon. I learned about this functionality from @adamjstewart.

Comment (alecbcs) [reply .github/dependabot.yml:15]: @alalazo are you worried about merging this without the feature being documented?

File: .github/workflows/valid-style.yml [hunk 3]
```diff
@@ -26,39 +26,43 @@ jobs:
...
       run: vermin --backport importlib --backport argparse --violations --backport typing -t=3.6- -vvv var/spack/repos
+
   # Run style checks on the files that have been changed
   style:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
       with:
         fetch-depth: 0
+        filter: tree:0
```
Comment (alalazo) [inline .github/workflows/valid-style.yml:42]: For the record, this line is the only functional change. The other changes are moving scripts into subfolders / renaming requirements.txt.

File: .github/dependabot.yml (hunk 2 above)
Comment (alalazo) [reply .github/dependabot.yml:15]: There's a conflict to be fixed. Otherwise this LGTM. The PR should probably mention reorganizing the placement of a few files, since it's 99% of what the PR does :wink:
